geneate_dynein_conformation_graphs/englishORspanish.py

```diff
@@ -0,0 +1,199 @@
+import numpy as np
+import os
+import pandas as pd
+import matplotlib.pyplot as plt
+
+# Declaring important folders
+rawdata_folder = 'Data'
+analysis_folder = 'Results'
+results_folder = "EORS"
+
+# Ensure output folder exists
+if not os.path.exists(f"{analysis_folder}/{results_folder}"):
+    os.makedirs(f"{analysis_folder}/{results_folder}")
+
+# Declare the results overview file and remove it if it already exists
+overview_results = "overview_bending_results.txt"
+overview_results_path = f"{analysis_folder}/{results_folder}/{overview_results}"
+
+if os.path.exists(overview_results_path):
+    os.remove(overview_results_path)
+
+# Adjustable parameters, print_results prints intermediate results into the console and show_plots displays the relevant plots while the code is running. This is useful for debugging for explaining what the code is doing.
+print_results = False
+show_plots = False
+
+# Function to calculate the angles between direction vectors of segments
+def calculate_angles(data, num_segments, angle_threshold):
+    # Calculate the length of each segment
+    segment_length = len(data) // num_segments
+
+    # Initialize empty lists
+    direction_vectors = [] # Initialize a list to store the direction vectors of each segment
+    angles = [] # Initialize a list to store the angles between consecutive direction vectors
+
+    # Loop through the data to create direction vectors for each segment
+    for i in range(num_segments):
+        # Extract the segment from the data
+        segment = data[i * segment_length:(i + 1) * segment_length]
+        
+        # Calculate the mean of the segment
+        segment_mean = segment.mean(axis=0)
+
+        # Perform Singular Value Decomposition (SVD) on the mean-centered segment
+        uu, dd, vv = np.linalg.svd(segment - segment_mean)
+        
+        # The first right singular vector (vv[0]) represents the direction of the segment
+        direction_vectors.append(vv[0])
+
+    # Loop through the direction vectors to calculate angles between consecutive vectors    
+    for i in range(len(direction_vectors) - 1):
+        vec1 = direction_vectors[i]
```

Comment:
this combo should almost always use enumerate
```suggestion
    for i, vec1 in enumerate(direction_vectors[:-1]):
```

Reply:
Although your code is a bit cleaner I would argue that my version is more readable. When you keep everything separate instead of combining random things, it is easier to interpret, especially when you are not familiar with the code.